sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_templates_and_assets.adoc

```diff
@@ -0,0 +1,308 @@
+product identifier as filename
+
+For example:
+```
+/assets/
+   asset-id-1/
+      PDS_OWASP_ZAP.zip
+      PDS_OWASP_ZAP.zip.sha256
+      OTHER_PRODUCT_X.zip
+      OTHER_PRODUCT_X.zip.sha256
+```
+[TIP]
+====
+The content and the structure inside the ZIP files is absolute solution specific :
+The PDS solution defines which kind of template data is necessary and how the 
+structure looks like!
+====
+
+===== Administration REST end points for storage
+Over REST API Administrators will be able to 
+
+- list (ZIP) file names 
+  `GET /api/administration/asset/$assetId/list`
+- upload asset ZIP files by REST API and create checksum file +
+  `PUT /api/administration/asset/$assetId/$productId`
+- download asset files by REST API  +
+  `GET /api/administration/asset/$assetId/$productId`
+- delete single asset file by REST API  +
+  `DELETE /api/administration/asset/$assetId/$productId`
+- delete all asset files by REST API  +
+  `DELETE /api/administration/asset/$assetId`
+
+The REST API will always create an audit log entry 
+
+[IMPORTANT]
+====
+This works only when {sechub} storage is reused by {pds}.
+If the {pds} uses its own storage (which should NOT be done for production, but only
+for PDS solution development) the assets must be uploaded directly to the PDS storage 
+location! 
+====
+
+===== SecHub template and asset handling
+====== Validation
+When a {sechub} job starts and there is a template data definition inside the configuration, 
+{sechub} will validate if the project has a template assigned for the location inside
+the configuration. e.g. templates with type `webscan-login` may only be defined inside 
+web scan login configuration).
+
+If this validation fails, the complete {sechub} job will fail and stop processing.
+
+[[sechub-concept-asset-upload-lazy]]
+====== Upload in storage as cache lazily
+When validation did not fail, {sechub} will check if the asset is already uploaded to
+storage (S3/NFS) already.
+
+When not uploaded to storage, the file will be uploaded before the job is further processed.
+
+====== Template PDS parameter calculation
+If validation and lazy uploaded did not fail, {sechub} will calculate a
+{pds} parameter : `pds.template.metadata` with JSON (see parameter syntax for details).
+
+.PDS parameter syntax for template meta data 
+[source,json]
+----
+{
+    "template-metadata" : [{//<1>
+        {
+            "template" : "single-singon", //<2>
+            "type: "webscan-login",//<3>
+            "assets" : [ //<4>
+                {
+                    "id" : "asset-id-1", //<5>
+                    "sha256" : "434c6c6ec1b0ed9844149069d7d45ac18e72505b" //<6> 
+                }
+            ]
+        }]
+    }     
+}
+
+----
+<1> Meta data array
+<2> Template identifier - just as an information for logging etc.
+<3> Template type
+<4> Asset information array
+<5> ID of the asset, necessary for storage download
+<6> checksum of the asset ZIP file, necessary to check after download
```

Comment:
In the example above there were multiple .zip files - each with its own checksum file.

What about just one .zip and checksum file per asset-id?
This would simplify things (also for the SecHub administrators)...